src/core/tests/pattern.cpp

```diff
@@ -484,6 +485,23 @@ TEST(pattern, matcher) {
     }
 }
 
+TEST(pattern, matching_optional) {
```

Comment:
can we check that "predicate" works for optional?
if "optional" pattern node does not match the criteria described in the predicate, then it is ignored and the comparison continues?

Reply:
applied